frontend/wasm/src/component/lift_exports.rs

```diff
@@ -81,8 +166,106 @@ pub fn generate_export_lifting_function(
         .map(|ba| ba as ValueRef)
         .collect();
 
-    // NOTE: handle CC lifting/lowering for non-scalar types
-    // see https://github.com/0xMiden/compiler/issues/369
+    // Export lifting: The core exported function returns a pointer to the result
+    // We need to:
+    // 1. Load the data from that pointer into the "flattened" representation (primitive types)
+    // 2. Return it as individual values (tuple)
+
+    let exec = fb.exec(core_export_func_ref, core_export_func_sig, args, span)?;
+
+    let borrow = exec.borrow();
+    let results = borrow.results().all();
+
+    // The core function should return a single pointer (as i32)
+    assert_eq!(results.len(), 1, "expected single result");
+    let result_ptr = results.into_iter().next().unwrap().borrow().as_value_ref();
```

Comment:
```suggestion
    let result_ptr = results[0].borrow().as_value_ref();
```

`EntityRange` implements `Index`, so you shouldn't have to convert it to an iterator just to get the first element.